app/src/main/java/com/easyfitness/DAO/DAOMachine.java

```diff
@@ -238,6 +250,18 @@ public void deleteAllEmptyExercises() {
         return getMachineList(selectQuery);
     }
 
+    /**
+     * @return List of Machine object ordered by Favorite and Name given exercise types as input
+     */
+    public ArrayList<Machine> getAllMachinesArray(String types) {
```

Comment:
As explained later, the parameters should rather be an array of ExerciseType.

Reply:
Explained above.